### PR TITLE
Device simulator dependency update round 2

### LINKED ISSUE #547: ‘Generate Code’ issue because azure-iothub-device-client lib？
It seems that azure-iothub-device-client has been replaced by iothub-client?
https://github.com/Azure/azure-iot-sdk-python/issues/270
When I follow the code generated by Azure IoT Toolkit extension to 'pip install azure-iothub-device-client', it's show not exist.

So if its correct, could you please update the 'generate code'?

### PR DESCRIPTION
**Pull request by c-ryan-k**
Java Device SDK update to v2.0.0
Python device sdk package migration to `azure-iot-device`, fixes #547 
Also adds explicit component governance reporting to our pipeline for builds without auto-injected policy.

### CODE REVIEW

File: resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java [hunk 1]
```diff
@@ -36,15 +37,16 @@ public String serialize() {
...
-  private static class EventCallback implements IotHubEventCallback {
-    public void execute(IotHubStatusCode status, Object context) {
-      System.out.println("IoT Hub responded to message with status: " + status.name());
-
-      if (context != null) {
-        synchronized (context) {
-          context.notify();
+  private static class EventCallback implements MessageSentCallback {
+    @Override
+    public void onMessageSent(Message sentMessage, IotHubClientException exception, Object callbackContext) {
```
Comment (vilit1) [inline resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java:42]: is there any need for sentMessage?

Comment (c-ryan-k) [reply resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java:42]: In our sample, no - but this is an implementation from `MessageSentCallback` - so the data will likely come from the SDK regardless - the user can check additional message properties or data in this callback.

**Review by vilit1 — APPROVED** [review]

File: resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java (hunk 1 above)
Comment (digimaun) [inline resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java:42]: Is there not a more specific type for callbackContext or is Object the most appropriate?

File: resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java [hunk 2]
```diff
@@ -36,15 +37,16 @@ public String serialize() {
...
+      System.out.println("IoT Hub responded to message with status: " + (exception == null ? IotHubStatusCode.OK.toString()
```
Comment (digimaun) [inline resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java:43]: Just an observation, this println to output the status code is more complicated than before.

File: resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java [hunk 3]
```diff
@@ -91,11 +93,11 @@ public void run() {
     }
   }
 
-  public static void main(String[] args) throws IOException, URISyntaxException {
+  public static void main(String[] args) throws IOException, URISyntaxException, IotHubClientException {
 
     // Connect to the IoT hub.
     client = new DeviceClient(connString, protocol);
-    client.open();
+    client.open(false);
```
Comment (digimaun) [inline resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java:100]: tiny nit, should a short code comment be added to explain client.open(false)?

**Review by digimaun — DISMISSED** [review]

File: resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java (hunk 1 above)
Comment (c-ryan-k) [reply resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java:42]: In this case, the only thing we're using the context object for is to manage the send/receive semaphore - I think an empty object works fine here but honestly I don't remember enough about Java to stand by that decision very strongly :)

**Review by digimaun — APPROVED** [review]

**Pull request merged**